### PR TITLE
Feature/picker date and emeralddatefield

### PR DESCRIPTION
**Pull request by cfpava**
fix bug year in emerald date field.

### CODE REVIEW

File: EmeraldIOS/Components/Date/EmeraldDateField.swift [hunk 1]
```diff
@@ -146,6 +147,7 @@ public class EmeraldDateField: EmeraldTextField, EmeraldDateFieldType, EmeraldDa
     }
     
     public func set(minimumDate: Date?) {
+        formatValidator.setDistanceDate(date: minimumDate!)
```
Comment (felipecondor) [inline EmeraldIOS/Components/Date/EmeraldDateField.swift]: Remove force unwrap

File: EmeraldIOS/Components/Date/EmeraldDateField.swift [hunk 2]
```diff
@@ -155,6 +157,7 @@ public class EmeraldDateField: EmeraldTextField, EmeraldDateFieldType, EmeraldDa
     
     public func set(maximumDate: Date?) {
         self.maximumDate = maximumDate
+        formatValidator.setDistanceDate(date: maximumDate!)
```
Comment (felipecondor) [inline EmeraldIOS/Components/Date/EmeraldDateField.swift]: remove force unwrap

File: ProofOfConcept/ProofOfConcept/ViewController.swift [hunk 3]
```diff
@@ -118,10 +117,8 @@ class ViewController: UIViewController, EmeraldValidableType {
         emeraldRegexFieldByStory.set(regex: .custom("^([A-Z0-9]{1}-[A-Z0-9]{5}-[A-Z0-9]{5}-[A-Z0-9]{5}-[A-Z0-9]{5})$"))
         emeraldSelectorByStory.set(notifiable: self)
         emeraldStartDateFieldByStory.setDependantField(with: emeraldEndDateFieldByStory)
-        emeraldStartDateFieldByStory.set(notifiable: emeraldStartDateFieldByStory)
-        emeraldStartDateFieldByStory.errorMessages.minimumDate = "This is a custom error"
-        emeraldStartDateFieldByStory.set(format: .shortDate)
-        emeraldStartDateFieldByStory.set(minimumDate: Date())
+        //emeraldStartDateFieldByStory.set(notifiable: emeraldStartDateFieldByStory)
+
```
Comment (felipecondor) [inline ProofOfConcept/ProofOfConcept/ViewController.swift]: please restore 

`emeraldStartDateFieldByStory.set(notifiable: emeraldStartDateFieldByStory)	        //emeraldStartDateFieldByStory.set(notifiable: emeraldStartDateFieldByStory)
        emeraldStartDateFieldByStory.errorMessages.minimumDate = "This is a custom error"	
        emeraldStartDateFieldByStory.set(format: .shortDate)	
        emeraldStartDateFieldByStory.set(minimumDate: Date())`

File: ProofOfConcept/ProofOfConcept/ViewController.swift [hunk 4]
```diff
@@ -119,9 +118,11 @@ class ViewController: UIViewController, EmeraldValidableType {
         emeraldSelectorByStory.set(notifiable: self)
         emeraldStartDateFieldByStory.setDependantField(with: emeraldEndDateFieldByStory)
         emeraldStartDateFieldByStory.set(notifiable: emeraldStartDateFieldByStory)
+        emeraldStartDateFieldByStory.set(notifiable: emeraldStartDateFieldByStory)
```
Comment (felipecondor) [inline ProofOfConcept/ProofOfConcept/ViewController.swift]: remove this line. Replicated line.

File: EmeraldIOS/Components/Date/FormatValidator.swift [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+//  FormatValidator.swift
+//  EmeraldIOS
+//
+//  Created by Carlos Pava on 10/15/19.
+//  Copyright © 2019 Condor Labs. All rights reserved.
+//
+
+import Foundation
+
+enum ComponentslongDate: Int {
```
Comment (mroncalloCondor) [inline EmeraldIOS/Components/Date/FormatValidator.swift]:
```suggestion
enum ComponentsLongDate: Int {
```

**Pull request merged**